specs/derivation.md

```diff
@@ -27,181 +43,606 @@ TODO
-- The L1 may [reorganize][g-reorg], requiring the submissions to reset to a previous point
-- The L2 grows, and does so in smaller intervals than L1
-- The L2 may have bursts that cannot be fully confirmed in a single piece of submitted data
+This means that even if the sequencer applies a state transition incorrectly, the transactions in the batch will stil be
+considered part of the canonical L2 chain. Batches are still subject to validity checks (i.e. they have to be encoded
+correctly), and so are individual transactions within the batch (e.g. signatures have to be valid). Invalid batches and
+invalid individual transactions within an otherwise valid batch are discarded by correct nodes.
 
-The multiplexed and buffered stream consists of:
+If the sequencer applies a state transition incorrectly, he will post an incorrect [output root][g-l2-output], which
+will be challenged by a [fault proof][g-fault-proof], then replaced by a correct output root **for the existing sequencer
+batches.**
 
-- L1 data transactions: any form of L1 data from which frames can be read, tagged with the L1 blockhash, number and timestamp.
-- Channels: independent of each other, a bounded stream submitted by a single actor, encoding a range of L2 bock derivation inputs.
-- Frames: numbered slices of channel data, which can be confirmed on L1 out of order and mixed with other channels.
+Refer to the [Batch Submission specification][batcher-spec] for more information.
 
-The types of data are illustrated in the diagram below.
+[batcher-spec]: batching.md
+
+> **TODO** rewrite the batch submission specification
+>
+> Here are some things that should be included there:
+>
+> - There may be different concurrent data submissions to L1
+> - There may be different actors that submit the data, the system cannot rely on a single EOA nonce value.
+> - The batcher requests batches from the rollup-node then builds channels and frames.
+
+## Batch Submission Wire Format
+
+[wire-format]: #batch-submission-wire-format
+
+Batch submission is closely tied to L2 chain derivation because the derivation process must decode the batches that have
+been encoded for the purpose of batch submission.
+
+The [batcher][g-batcher] submits [batcher transactions][g-batcher-transaction] to a [data availability
+provider][g-avail-provider]. These transactions contain one or multiple [channel frames][g-channel-frame], which are
+chunks of data belonging to a [channel][g-channel].
+
+A [channel][g-channel] is a sequence of [sequencer batches][g-sequencer-batch] (for sequential blocks) compressed
+together. The reason to group multiple batches together is simply to obtain a better compression rate, hence reducing
+data availability costs.
+
+Channels might be too large to fit in a single [batcher transaction][g-batcher-transaction], hence we need to split it
+into chunks known as [channel frames][g-channel-frame]. A single batcher transaction can also carry multiple frames
```

Comment:
The way I see it, allowing this gives us maximum flexibility, and ease of implementation (since we do not need to check the constraint of one frame per channel).

I can think of cases where this could be desirable too, e.g. if we limit frame size but data availability space is pretty big, or just let the batcher schedule things according to its own logic (say he hasn't seen new things in a while, he does a frame, but then suddenly new things come in, and can just put those in another frame).